src/Microsoft.Deployment.DotNet.Releases/src/ReleaseVersion.cs

```diff
@@ -568,5 +559,281 @@ internal static int CompareNumericIdentifier(string a, string b)
+
+        /// <summary>
+        /// Converts the string representation of a version to an equivalent <see cref="ReleaseVersion"/> object.
+        /// </summary>
+        /// <param name="input">A string that contains a version to convert.</param>
+        /// <returns>An object that is equivalent to the version number specified in the <paramref name="input"/> parameter.</returns>
+        /// <exception cref="ArgumentNullException" />
+        /// <exception cref="FormatException" />
+        public static ReleaseVersion Parse(string input)
+        {
+            if (input == null)
+            {
+                throw new ArgumentNullException(nameof(input));
+            }
+
+            if (!TryParse(input, canThrow: true, out ReleaseVersion parsedVersion))
+            {
+                throw new FormatException(ReleasesResources.InvalidReleaseVersion);
+            }
+
+            return parsedVersion;
+        }
+
+        /// <summary>
+        /// Tries to convert the string representation of a version to an equivalent <see cref="ReleaseVersion"/> object,
+        /// and returns a value indicating whether the conversion succeeded.
+        /// </summary>
+        /// <param name="input">A string that contains a version to convert.</param>
+        /// <param name="result">Contains the <see cref="ReleaseVersion"/> equivalent of the value in <paramref name="input"/> if
+        /// successful; otherwise <see langword="null"/>.</param>
+        /// <returns><see langword="true"/> if the string could be converted; <see langword="false"> otherwise.</see> </returns>
+        public static bool TryParse(string input, out ReleaseVersion result)
+        {
+            return TryParse(input, canThrow: false, out result);
+        }
+
+        /// <summary>
+        /// Determines whether a string is a valid numeric identifier.
+        /// </summary>
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string is a numeric identifier; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsNumericIdentifier(string input)
+        {
+            // <numeric identifier> ::= "0" | <positive digit> | <positive digit> <digits>
+            if (string.IsNullOrWhiteSpace(input) || (input[0] == '0' && input.Length > 1))
+            {
+                return false;
+            }
+
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (!char.IsDigit(input[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines if a string only contains digits.
+        /// </summary>
+        /// <param name="input">The string to check.</param>
+        /// <returns><see langword="true"/> if the string only contains digits; <see langword="false"> otherwise.</see></returns>
+        internal static bool IsDigits(string input)
```

Comment:
Consider renaming to something like IsAllDigits.  To me IsDigits implies it has to have more than one digit in order to return true.